beetsplug/export.py

```diff
@@ -108,44 +135,118 @@ def run(self, lib, opts, args):
             except (mediafile.UnreadableFileError, IOError) as ex:
                 self._log.error(u'cannot read file: {0}', ex)
                 continue
-
             data = key_filter(data)
             items += [data]
 
-        export_format.export(items, **format_options)
+        self.run_results = items
+        self.export_format.export(self.run_results, **format_options)
```

Comment:
As mentioned above, I think it would be awesome to remove these side effects on mutable class fields…

Reply:
will do